utils/resource/usage.go

```diff
@@ -143,22 +185,47 @@ func (m *manager) update(diskPath string, frequency, cpuHalflife, diskHalflife t
 
 	frequencyInSeconds := frequency.Seconds()
 	for {
-		currentCPUUsage, currentReadUsage, currentWriteUsage := m.getActiveUsage(frequencyInSeconds)
+		currentCPUUsage, currentMemoryUsage, currentReadUsage, currentWriteUsage := m.getActiveUsage(frequencyInSeconds)
 		currentScaledCPUUsage := newCPUWeight * currentCPUUsage
 		currentScaledReadUsage := newDiskWeight * currentReadUsage
 		currentScaledWriteUsage := newDiskWeight * currentWriteUsage
 
+		machineMemory, getMemoryErr := mem.VirtualMemory()
+		if getMemoryErr != nil {
+			m.log.Debug("failed to lookup resource",
+				zap.String("resource", "system memory"),
+				zap.Error(getMemoryErr),
+			)
+			machineMemory = &mem.VirtualMemoryStat{}
+		}
+		machineSwap, getSwapErr := mem.SwapMemory()
+		if getSwapErr != nil {
+			m.log.Debug("failed to lookup resource",
+				zap.String("resource", "system swap"),
+				zap.Error(getSwapErr),
+			)
+			machineSwap = &mem.SwapMemoryStat{}
+		}
 		availableBytes, getBytesErr := storage.AvailableBytes(diskPath)
+		if getBytesErr != nil {
+			m.log.Debug("failed to lookup resource",
+				zap.String("resource", "system disk"),
+				zap.Error(getBytesErr),
+			)
+		}
 
 		m.usageLock.Lock()
 		m.cpuUsage = oldCPUWeight*m.cpuUsage + currentScaledCPUUsage
+		m.memoryUsage = currentMemoryUsage
 		m.readUsage = oldDiskWeight*m.readUsage + currentScaledReadUsage
 		m.writeUsage = oldDiskWeight*m.writeUsage + currentScaledWriteUsage
 
+		if getMemoryErr == nil {
```

Comment:
nit: may want to comment that it is ok if `getSwapErr != nil` here so it doesn't look like a missed check